### ISSUE TITLE
Google will be blocking openWmail March 31st, 2019

### ISSUE THREAD

**Issue by ronenrozner** [openWMail/openWMail#63]
* Which version of openWMail are you using? current

* Which Operating System are you using? Any

From an email I got today:
_"we wanted to let you know that the following third-party apps will no longer be able to access some data in your Google Account, including your Gmail content. This change will go into effect starting March 31, 2019.

openWMail
We are making this change as part of ongoing efforts to make sure your data is protected and private. These apps haven’t yet complied with our [updated data privacy requirements](https://www.google.com/appserve/mkt/p/AFnwnKW5eHQxK_nKfC_ZIt_Sv4ngjACWvoopXEiu9hlpyphFPmf3yzK6qY0omcOBROWYDLFJLpksMcmydhExBfIusEX7JI3Hp9UlhgzE3iCvpqw6C1vc9_uAj1s2Q6M6S2h5M5uAcHxcr5ZOkXEleeiYUgpN166cbLQWBmMZa6Q) announced on October 8, 2018."_

Will there be an update to the app to make it compatible?

**Comment by natedennis** [comment]
Here searching for the same answer.

**Comment by geekgonecrazy** [comment]
Some how seems like the email went to the wrong email.  :( I've submitted to google for review.  Seems they don't like that we use the read scope.  So have to explain why we use it for notifications / unread counts.

🤞 hopefully they review quickly

**Comment by luzpaz** [comment]
https://venturebeat.com/2019/04/18/google-will-begin-to-block-sign-ins-from-embedded-browser-frameworks-in-june/

**Comment by geekgonecrazy** [comment]
I have an appeal still going.  Got a reply today that acts like they might be missing something.  But only thing I can tell is they mentioned an optional youtube video demoing the oauth login.  

Hopefully they move this forward and don't let that hang our review up.  @WillPresley I don't know if you have any thoughts here?


btw just throwing this out there, if any others have experience here we are definitely looking for others to help maintain this project.  I personally have stopped using this as much since the inbox product went away.

**Comment by WillPresley** [comment]
I've been keeping an eye on this, but my company recently spun up a retail business so I've had even less time than normal for my passion projects. @geekgonecrazy I'd love to hear what they have to say, once they get back to you. I have no real thoughts, aside from echoing your words on looking for more maintainers to help us out.

For what it's worth, I am trying out WaveBox (the long-ago forked ancestor to Wmail/openWmail, for those who don't remember), and I'll be damned if it's not a very nice product. $20/year nice? We'll see I guess. 

I think this project only really lives on if we get past this Google/Gmail verification process. **If we do**, the first priority for us (and any potential additional maintainers) would be to remove all references to the Inbox product. At that point, this essentially becomes a Gmail/Calendar/Contacts wrapper app, which is an interesting/boring thought, and why both Wavebox and Rambox have gone the 'add whatever services' route.

**Comment by natedennis** [comment]
boring as it may be. a working wrapper with desktop notifications without
an additional tab in my 40+ tab browser running on a linux platform has
significant worth to me. thunderbird is a complete POS though i have used
it for years i really dont want to go back to it. this project is written
in the wrong language for me to be a lot of help, but that said, i'll help
where i can.

**Comment by geekgonecrazy** [comment]
So far the communications I've gotten are:

![Google OAuth Review Request](https://user-images.githubusercontent.com/51996/56617281-1eca4780-65e5-11e9-9c8a-d8743b859862.png)

Then:

<img width="705" alt="image" src="https://user-images.githubusercontent.com/51996/56617303-2ee22700-65e5-11e9-9af8-d3c07aba2bf6.png">

This one is a bit odd.. because its acting like they sent some request for info.   I can't find any such request for info.

From the actual oauth section in google I seee:

<img width="489" alt="image" src="https://user-images.githubusercontent.com/51996/56617398-65b83d00-65e5-11e9-8576-874f85fefb2c.png">

and

<img width="478" alt="image" src="https://user-images.githubusercontent.com/51996/56617437-7d8fc100-65e5-11e9-8067-5291b6d7f006.png">

Odd that they force the authorized domain.. but wouldn't allow the required fields below it to be filled in with out it.

But can find no other correspondence.. 

I believe the read scope is only used for notifications.  Which is one of the key aspects of this.. but maybe we can find a different scope that will still allow the api calls to work?

**Comment by geekgonecrazy** [comment]
https://github.com/openWMail/openWMail/blob/master/src/scenes/mailboxes/src/stores/google/googleHTTP.js#L165

From quick look its primarily the getThread I think that needs this scope.  I'll have to do more digging to see how things will behave if this is unable to get a result.  We might be able to remove some behavior if they deny us

**Comment by natedennis** [comment]
welp they finally got us... i can nolonger login from openwmail.

**Comment by babinux** [comment]
Unable to add a new account. 
I hope you can find a fix, I love this app 🥰

<img width="912" alt="Screen Shot 2019-08-13 at 4 20 18 PM" src="https://user-images.githubusercontent.com/2189013/62984010-5f450f80-bde6-11e9-97a1-b4af4c993f90.png">

**Comment by geekgonecrazy** [comment]
Yeah they have rejected the appeal :( so we will have to reduce the scopes we request and have some broken functionality.

**Comment by c-m-a-n** [comment]
This has been an ideal application for the past 2 years. Just perfect! I rely on this application daily!
...and now we face this auth issue with Google.

> Yeah they have rejected the appeal :( so we will have to reduce the scopes we request and have some broken functionality.

Any idea:

1) What type of functionality we may have to lose?
2) How long the re-submission and subsequent approval from Google will take?

**Comment by geekgonecrazy** [comment]
I spent a few hours trying to find a way to make it usable with out this scope.  Not possible.  I've submitted the request once again to have this verified and included this issue in the request

**Comment by c-m-a-n** [comment]
> I spent a few hours trying to find a way to make it usable with out this scope. Not possible. I've submitted the request once again to have this verified and included this issue in the request

Thanks so much @geekgonecrazy  If its not possible to use the application without this scope, I fear the end is nearly here.... such a shame! This app was/is awesome. 

Are there any alternatives people are using? Much prefer the wrapper style 'lite' client, rather than a fully fledged service (that have their own servers, UI, etc.) especially in today's challenges regarding data privacy.

**Comment by WillPresley** [comment]
Hey Aaron/ @geekgonecrazy just wanted to give my thanks for your work on trying to get this all sorted out. I've followed from afar, as there isn't much else to do but hope Google check out this repo and change their minds, but if we can get approved, I plan to jump back into this project fairly heavily.

As for alternatives, I kept it 'in the family' and gave Wavebox (what Wmail became, most of openWmail was written by Thomas Beverley) a shot. It's honestly very well made, but I still yearn for something that is free for a bunch of accounts and solely for Google services.

**Comment by c-m-a-n** [comment]
> but I still yearn for something that is free for a bunch of accounts and solely for Google services.

Ditto. Multiple account support is the killer feature. Fingers crossed!!

**Comment by geekgonecrazy** [comment]
> Hey Aaron/ @geekgonecrazy just wanted to give my thanks for your work on trying to get this all sorted out. I've followed from afar, as there isn't much else to do but hope Google check out this repo and change their minds, but if we can get approved, I plan to jump back into this project fairly heavily.

Yes unfortunately its in their hands.  I figured that the initial appeal would have been approved.  A quick look or try of it is pretty obvious that its an email client.

> but I still yearn for something that is free for a bunch of accounts and solely for Google services.

Yes there is something about the dedicated client just for email / calendar / tasks that is very nice.  

There are many tools out there that do the multi-apps in a box thing like wavebox.  I feel what they are doing is not unique there at all.  

https://rambox.pro/#home is one I had used in the past before I even came across wmail.  But moved to wmail because of its simplicity and focus on just email.

🤞we'll reach a human in Google that will give this a look.

**Comment by natedennis** [comment]
Ive been looking at rambox.. things i dont like is the tabs and the
notifications dont really work so well. Clicking on them does nothing in
gnome 3. I really just want this to work.

**Comment by c-m-a-n** [comment]
> 🤞we'll reach a human in Google that will give this a look.
Do we have an ETA at all? Have zero knowledge of the lead times involved for Google to re-review something like this....

**Comment by geekgonecrazy** [comment]
They gave very generic time table unfortunately

**Comment by c-m-a-n** [comment]
Any hope @geekgonecrazy ?

**Comment by geekgonecrazy** [comment]
Haven't heard a word.  :(  if anyone else is interested in helping me push this forward please feel free to reach out.

Can be found at: geekgonecrazy on keybase chat / protonmail / telegram etc.

**Comment by luzpaz** [comment]
what are other alternatives in the meantime that people are using ?

**Comment by ronenrozner** [comment]
> what are other alternatives in the meantime that people are using ?

Although openWmail is not getting authorized, I can still use it to access my email and the other Google services. So it's still functioning for me, and I didn't really replace it.

**Comment by luzpaz** [comment]
> Although openWmail is not getting authorized, I can still use it to access my email and the other Google services. So it's still functioning for me, and I didn't really replace it.

It doesn't even start for me anymore.

**Comment by c-m-a-n** [comment]
openWMail has really started to falter now (as of Aug 2020) - seeing multiple authentication issues across accounts repeatedly - and now the app itself fails to open at times. On the occasions it does work, add attachments to emails doesn't work, so it's remains a compromised user experience. 

In summary then, it looks like it's days are numbered, unless there is any hope via: https://github.com/openWMail/openWMail/issues/83

In the absence of any decent lightweight alternative, am considering using something like FluidApp (may no longer be maintained) or Coherence X or Unite which allow you to create 'apps from web pages'. Not ideal, but essentially serve the same purpose: a wrapper for Gmail that can sit in your dock.

Has anyone else found alternatives?
